tools/mc-java/src/main/java/io/spine/tools/mc/java/codegen/MessagesConfig.java

```diff
@@ -0,0 +1,137 @@
+import static io.spine.tools.mc.java.codegen.Names.className;
+import static java.util.stream.Collectors.toSet;
+
+/**
+ * A codegen configuration for messages which match a certain pattern.
+ *
+ * @see Codegen#forMessages
+ */
+public final class MessagesConfig extends ConfigWithFields<ForMessages> {
+
+    private final Pattern pattern;
+    private final SetProperty<String> methodFactories;
+    private final SetProperty<String> nestedClassFactories;
+
+    MessagesConfig(Project p, Pattern pattern) {
+        super(p);
+        this.pattern = pattern;
+        methodFactories = p.getObjects()
+                           .setProperty(String.class)
+                           .convention(ImmutableList.of());
+        nestedClassFactories = p.getObjects()
+                                .setProperty(String.class)
+                                .convention(ImmutableList.of());
+    }
+
+    void emptyByConvention() {
+        interfaceNames().convention(ImmutableSet.of());
+        methodFactories.convention(ImmutableSet.of());
+        nestedClassFactories.convention(ImmutableSet.of());
+    }
+
+    /**
+     * Specifies a {@link io.spine.tools.protoc.MethodFactory} to generate methods for
+     * the message classes.
+     *
+     * <p>Calling this method multiple times will add provide factories for code generation.
+     *
+     * @param factoryClassName
+     *         the canonical class name of the method factory
+     */
+    public void generateMethodsWith(String factoryClassName) {
+        methodFactories.add(factoryClassName);
+    }
+
+    /**
+     * Specifies a {@link io.spine.tools.protoc.NestedClassFactory} to generate nested classes
+     * inside the message classes.
+     *
+     * <p>Calling this method multiple times will add provide factories for code generation.
+     *
+     * @param factoryClassName the canonical class name of the method factory
+     */
+    public void generateNestedClassesWith(String factoryClassName) {
+        nestedClassFactories.add(factoryClassName);
+    }
+
+    @Override
+    ForMessages toProto() {
+        ForMessages.Builder result = ForMessages.newBuilder()
+                .setPattern(pattern)
+                .addAllAddInterface(interfaces())
+                .addAllGenerateMethods(generateMethods())
+                .addAllGenerateNestedClasses(generateNestedClasses());
+        GenerateFields generateFields = generateFields();
+        if (isNotDefault(generateFields)) {
+            result.setGenerateFields(generateFields);
+        }
+        return result.build();
+    }
+
+    private Set<GenerateMethods> generateMethods() {
+        return methodFactories.get()
+                              .stream()
+                              .map(name -> MethodFactoryName.newBuilder()
+                                      .setClassName(className(name))
+                                      .build())
+                              .map(name -> GenerateMethods.newBuilder()
```

Comment:
Please extract a method which does the `name` -> `MethodFactoryName` -> `GenerateMethods` conversion. This way we'd have only one `map`, and it would be easier to understand.